src/Riok.Mapperly/Descriptors/Mappings/ForEachAddEnumerableMapping.cs

```diff
@@ -14,24 +14,50 @@ namespace Riok.Mapperly.Descriptors.Mappings;
 public class ForEachAddEnumerableMapping : ExistingTargetMappingMethodWrapper
 {
     private readonly ObjectFactory? _objectFactory;
+    private readonly ITypeSymbol _typeToInstantiate;
+    private readonly string? _countPropertyName;
 
+    // warning: if countPropertyName is not null then typeToInstantiate must have a single integer constructor
```

Comment:
This comment applies to the second ctor but not to this one, does it? I'd also make it an xml comment to show it up in the IDE when invoking the constructor and reference the parameters with paramref.